This is 07-fix-includes.patch from Debian and linux-igd-includes.patch from Fedora for bug 401107.

--- gatedevice.c
+++ gatedevice.c
@@ -1,4 +1,5 @@
 #include <syslog.h>
+#include <stdio.h>
 #include <stdlib.h>
 #include <upnp/ixml.h>
 #include <string.h>
--- main.c
+++ main.c
@@ -3,8 +3,8 @@
 #include <unistd.h>
 #include <syslog.h>
 #include <signal.h>
+#include <string.h>
 #include <sys/types.h>
-#include <unistd.h>
 #include <sys/stat.h>
 #include <sys/resource.h>
 #include <time.h>
--- pmlist.c
+++ pmlist.c
@@ -1,4 +1,6 @@
 #include <stdlib.h>
+#include <stdio.h>
+#include <string.h>
 #include <sys/wait.h>
 #include <unistd.h>
 #include <upnp/upnp.h>
